src/newscale/device_codes.py

```diff
@@ -0,0 +1,183 @@
+"""Collection of interface-agnostic device commands to get/set state."""
+
+import logging
+from bitstring import BitArray
+import sys
+from enum import Enum
+
+try:
+    from enum import StrEnum  # a 3.11 freature.
+except ImportError:
+    logging.error("StrEnum not available with this version of python. "
+                  "Creating it.")
+
+    class StrEnum(str, Enum):
+        pass
+
+
+# Constants
+TRANSCEIVER_PREFIX = "TR"
+
+
+# Commands, collected as Enums
+class StageCmd(StrEnum):
```

Comment:
same as below, what stages do these apply to? probably not all newscale stages. therefore they should be scoped to M3-LS-3.4-15 or similar.

Reply:
Generally, there's M3-LS, M3-RS, and M3-FS.

M3LinearSmartStage is the M3-LS. Happy to rename that to M3LS if needed, but "Linear Smart Stage" was the same thing Newscale called them and felt more descriptive.

For the specific model types (M3-LS-3.4-15, etc.), the commands are actually identical, although in like one case they have a different response. In that case, we'd want to make a specific `StageReplyEncoding` dict for it.